base/compiler/ssair/ir.jl

```diff
@@ -1469,62 +1466,104 @@ function maybe_erase_unused!(
     return false
 end
 
-function fixup_phinode_values!(compact::IncrementalCompact, old_values::Vector{Any})
+struct FixedNode
+    node::Any
+    needs_fixup::Bool
+    FixedNode(@nospecialize(node), needs_fixup::Bool) = new(node, needs_fixup)
+end
+
+function fixup_phinode_values!(compact::IncrementalCompact, old_values::Vector{Any}, reify_new_nodes::Bool)
     values = Vector{Any}(undef, length(old_values))
+    needs_fixup = false
     for i = 1:length(old_values)
         isassigned(old_values, i) || continue
         val = old_values[i]
-        if isa(val, Union{OldSSAValue, NewSSAValue})
-            val = fixup_node(compact, val)
+        if isa(val, OldSSAValue)
+            val = compact.ssa_rename[val.id]
+            if isa(val, SSAValue)
+                compact.used_ssas[val.id] += 1
+            end
+        elseif isa(val, NewSSAValue)
+            if reify_new_nodes
+                val = SSAValue(length(compact.result) + val.id)
+            else
+                needs_fixup = true
+            end
         end
         values[i] = val
     end
-    values
+    return FixedNode(values, needs_fixup)
```

Comment:
A usage of `FixedNode` here seems weird conceptually. It would be more natural to just return a tuple here.